packages/components/src/components/DistributionChart.tsx

```diff
@@ -23,29 +24,88 @@ export const DistributionChart: React.FC<DistributionChartProps> = ({
   height,
   width,
 }: DistributionChartProps) => {
+  let [isLogX, setLogX] = React.useState(false);
+  let [isExpY, setExpY] = React.useState(false);
   const [sized, _] = useSize((size) => {
     let shape = distribution.pointSet();
-    let widthProp = width !== undefined ? width - 20 : size.width - 10;
     if (shape.tag === "Ok") {
-      return (
-        <div>
-          <SquiggleVegaChart
-            data={{ con: shape.value.continuous, dis: shape.value.discrete }}
-            width={widthProp}
-            height={height}
-            actions={false}
-          />
-        </div>
-      );
+      let spec = buildSpec(isLogX, isExpY, shape.value);
+      if (spec.tag == "Ok") {
+        let widthProp = width ? width - 20 : size.width - 10;
+        var result = (
+          <div>
+            <Vega
+              spec={spec.value}
+              data={{ con: shape.value.continuous, dis: shape.value.discrete }}
+              width={widthProp}
+              height={height}
+              actions={false}
+            />
+          </div>
+        );
+      } else {
+        var result = (
+          <ErrorBox heading={spec.value.heading}>{spec.value.error}</ErrorBox>
+        );
+      }
     } else {
-      return (
-        <div>
-          <ErrorBox heading="Distribution Error">
-            {distributionErrorToString(shape.value)}
-          </ErrorBox>
-        </div>
+      var result = (
+        <ErrorBox heading="Distribution Error">
+          {distributionErrorToString(shape.value)}
+        </ErrorBox>
       );
     }
+    return (
+      <>
+        {result}
+        <div>
+          <CheckBox label="Log X scale" value={isLogX} onChange={setLogX} />
+          <CheckBox label="Exp Y scale" value={isExpY} onChange={setExpY} />
+        </div>
+      </>
+    );
   });
   return sized;
 };
+
+type ViewError = { heading: string; error: string };
+
+function buildSpec(
+  isLogX: boolean,
+  isExpY: boolean,
+  shape: shape
+): result<VisualizationSpec, ViewError> {
+  let someBelow0 =
+    shape.continuous.some((x) => x.x <= 0) ||
+    shape.discrete.some((x) => x.x <= 0);
+  if (!(isLogX && someBelow0)) {
+    return {
+      tag: "Ok",
+      value: {
+        ...chartSpecification,
+        scales: [
+          isLogX ? logXScale : linearXScale,
+          isExpY ? expYScale : linearYScale,
+        ],
+      } as VisualizationSpec,
+    };
+  } else {
+    return {
+      tag: "Error",
+      value: {
+        heading: "Log Viewing error",
+        error:
+          "Distribution contains values lower than or equal to 0. Cannot view",
```

Comment:
Do we need this, now that we use ``isExpY``?

Reply:
This is for x coordinates and not y coordinates. X used to use exponential scaling, but it had some really awkward results like scrunching up the values on the x axis.

Reply:
If you'd prefer allowing values lower than 0 over having a real log x scale, then I can change it back

Reply:
Separate:
How about the message,
"The distribution contains values at or below 0. Logarithm only works on positive values."
Or similar